app/scripts/controllers/buildConfig.js

```diff
@@ -65,6 +66,40 @@ angular.module('openshiftConsole')
               });
             }
 
+            $scope.updatedBuildConfig = angular.copy($scope.buildConfig);
+            $scope.buildStrategy = $filter('buildStrategy')($scope.updatedBuildConfig);
+            $scope.strategyType = $scope.buildConfig.spec.strategy.type;
+            if(!$scope.buildStrategy.env) {
+              $scope.buildStrategy.env = [];
+            }
+            $scope.envVars = $scope.buildStrategy.env;
+
+            $scope.saveEnvVars = function() {
+              $scope.envVars = _.reject(  $scope.envVars, function(env) {
+                return env.name === "";
```

Comment:
any reason not to have `return !env.name`  that way if anything ever changed and you got back null / undefined name then it would still reject it